src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -255,7 +277,7 @@ private BoundForEachStatement BindForEachPartsWorker(DiagnosticBag diagnostics,
             hasErrors = hasErrors || iterationVariableType.IsErrorType();
 
             // Skip the conversion checks and array/enumerator differentiation if we know we have an error (except local name conflicts).
-            if (hasErrors)
+            if (hasErrors || _syntax.HasErrors)
```

Comment:
I'm fine with suppressing semantic errors in the face of syntax errors, given that we've bound all the pieces of the syntax needed for a good IDE/SemanticModel experience.

---
In reply to: [89846888](https://github.com/dotnet/roslyn/pull/15548#discussion_r89846888) [](ancestors = 89846888)